### ISSUE TITLE
Court order pages: add court order information

### ISSUE THREAD

**Issue by JimKillock** [openrightsgroup/blocked-org-uk#139]
Under the administration system, allow an admin user to add:

(1) A Court Order: consisting of a Name, Date and URL; optionally: 

(a) A court judgment, date and URL

(2) Allow URLs blocked by that Court Order to be assigned to the Court Order

(3) Display Court Order information on the Court Order page. Order page blocks by Court Order, then by URL (group all ISP blocks per URL)

(4) Include judgment link where present

**Comment by JimKillock** [comment]
PS: we will need this feature or something similar per country for the EU website. So we'd sometime have to think about whether this is all done in a custom manner, as present.

**Comment by dantheta** [comment]
1 & 2 are now present in the control panel.  Splitting the court order pages changes  to a separate ticket.

**Comment by dantheta** [comment]
To double-check, for 1a, should there be three additional fields (judgement, judgement_date, judgement_url)?

**Comment by JimKillock** [comment]
Yes, that is correct. Note, for legal judgments, it is “judgment” and not “judgement”

**Comment by JimKillock** [comment]
I've made a bit of a mistake here. The court orders are documents per ISP, so the structure should be:

(1) A "Legal block": consisting of a Name, Date and wiki URL:

(a) A court judgment, date and URL

(b) Multiple Court Order URLs, each assigned to an ISP

**Comment by dantheta** [comment]
The court order editor in the control panel is mostly complete.  There's an edit screen that isn't there just yet (which is where the details of the court order record can be edited and the per-ISP info URLs can be added).  That's coming in the next batch.

**Comment by JimKillock** [comment]
Great, thanks Dan

**Comment by dantheta** [comment]
Adding, deleting and editing court orders is now supported, as well as adding ISP court order info (that's on the edit screen).  Adding blocked URLs to the court orders works too.  It's very fussy about the blocked URLs - we can make the UI better for that (such as allowing admins to add blocked sites to a court order from the public frontend, where it's easier to discover/browse URLs).

**Comment by JimKillock** [comment]
Great, thanks Dan. One question and some queries:

(1) Each ISP receives a separate court order. Is this what we mean by "ISP info URL"?

(2) Each court order may affect multiple websites, eg piratebay.org plus torrentsite.org — with each website that is affected, there are as you know multiple domains serving each, eg piratebayclone.org, plus torrentsiteclone.org identified in an ad hoc manner later.

We should therefore add a further layer:

(a) Website blocked by court order

and assign the URLs to (2) (a) so they can be grouped by website (eg piratebay.org group includes piratebayclone.org; with a further group for torrentsite.org and torrentsiteclone.org )

rather than to the court order (which would include piratebay.org plus torrentsite.org and piratebayclone.org plus torrentsiteclone.org )

**Comment by JimKillock** [comment]
Further checks on court orders, eg http://www.bailii.org/ew/cases/EWHC/Ch/2013/3479.html . They have three standard references:

(1) Name, eg "Paramount Home Entertainment International Ltd & Ors v British Sky Broadcasting Ltd & Ors"

(2) Citation (which I think refers to the judgment, eg [2013] EWHC 3479 (Ch) 

(3) Case number, eg HC13F04302

Can we add (2) and (3) and a further field:

(4) Sites or content targeted: a long freeform field to explain which websites are targeted, eg in this case:

www.solarmovie.so ("SolarMovie") 
www.tubeplus.me ("TubePlus")

**Comment by gwire** [comment]
Note that you may not know any of the details except the descriptive names of the sites being blocked, so we need to create persistent references (e.g. for URLs), and not use the court's.

Most judgements for S97A cases are not published, and there doesn't seem to be a way to discover the details other than inferring them from the slightly obfuscated details the ISPs publish.

See the cases on https://wiki.451unavailable.org.uk/

**Comment by JimKillock** [comment]
@gwire yes, currently we include the wiki.451 URLs in the backend for this reason. I hope we can find a way to make the information more complete tho.

**Comment by dantheta** [comment]
I'm preparing a second iteration of the data model for the court orders.  I think the first pass had inverted court order and court judgment.  I've also thought up a much more convenient storage method for these that will cut the development time and scope for errors.

**Comment by JimKillock** [comment]
Thanks Dan. I think that's right, which is a result of my / the world in general's sloppy use of language and "court order" we generally have about this topic.

**Comment by dantheta** [comment]
It's all good :)

**Comment by JimKillock** [comment]
In the light of the UGG seizure record, we could add:

(0) Type: Seizure / Block / Content removal / Other
(1) Jurisdiction (Country) (state)
(2) Court where order obtained
(3) Order obtained by
(4) Represented by
(5) Other court documents (free form, as HTML?)

**Comment by JimKillock** [comment]
As this has got long and complicated I've summarised all the info and fields needed here:

https://wiki.451unavailable.org.uk/wiki/UK_blocking_injunction_template

**Comment by JimKillock** [comment]
Minor tidyups:

(0) On the admin list of injunctions, because the title does not include the names of the sites, display the "sites affected" field underneath the title.

(1) Rename the Court Order Admin pages as "Legal requests" "Legal requests Admin"

(2) Reword "Injunction obtained by" to "Application made by"

(3) Reword "Injunction represented by" to "Applicants  represented by"

(4) For each ISP affected + Court order URL add a date option

(5) For the injunctions admin page, order by date

The next bit requires some thought. Is is better to represent *domain seizures* or *suspensions* separately in the database? For instance, as they don't apply to ISPs, but are global in nature, and could be voluminous?

If they stay in the same system, then we would add in options for "applies to" to include

* Domain provider (blank field?)
* Domain registry (list?)

**Comment by JimKillock** [comment]
All current orders we know about are now entered into the system.

**Comment by dantheta** [comment]
1-3 are done.

For the suspensions, I think it might be worth storing them separately.  As you say, the available information is quite differently shaped, and we'd also want to display a seized domain's status differently.

**Comment by JimKillock** [comment]
OK. I've added a separate suspensions / seizures ticket at #282, and a further ticket about multiple sets of documents at #285 which we can deal with in this round of changes, or later, depending on your preference.

**Comment by JimKillock** [comment]
Can we do the remaining parts:

(0) On the admin list of injunctions, because the title does not include the names of the sites, display the "sites affected" field underneath the title.

(4) For each ISP affected + Court order URL add a date option

(5) For the injunctions admin page, order by date

ASAP to help @ajhaydock with the court order info (as he requests court orders we will want to keep our data as consistent and accurate as possible?

**Comment by dantheta** [comment]
Added sites affected text and changed the sort order.

**Comment by dantheta** [comment]
Date field added (4)

**Comment by JimKillock** [comment]
I think this is done then and can be closed. There's further issues at  #282, and at #285 for other changes. Thanks Dan!
